plugins/document-resources/src/components/navigator/TeamspaceSpacePresenter.svelte

```diff
@@ -125,6 +149,104 @@
 
     return result
   }
+
+  let draggedItem: Ref<Document> | undefined = undefined
+  let draggedOver: Ref<Document> | undefined = undefined
+  let draggedOverBefore = false
+  let draggedOverAfter = false
+  let cannotDropTo: Ref<Document>[] = []
+
+  function canDrop (object: Ref<Document>, target: Ref<Document>): boolean {
+    if (object === target) return false
+    if (cannotDropTo.includes(target)) return false
+
+    return true
+  }
+
+  function onDragStart (event: DragEvent, object: Ref<Document>): void {
+    // no prevent default to leverage default rendering
+    // event.preventDefault()
+    if (event.dataTransfer === null) {
+      return
+    }
+
+    const doc = documentById.get(object)
+    if (doc === undefined) {
+      return
+    }
+
+    cannotDropTo = [doc._id, doc.attachedTo, ...getAllDescendants(object)]
+
+    event.dataTransfer.effectAllowed = 'move'
+    event.dataTransfer.dropEffect = 'move'
+    draggedItem = object
+
+    closeTooltip()
+  }
+
+  function getDropPosition (event: DragEvent): { before: boolean, after: boolean } {
+    const targetRect = (event.target as HTMLElement).getBoundingClientRect()
+    const dropPosition = event.clientY - targetRect.top
+    return {
+      before: dropPosition < targetRect.height / 6,
+      after: dropPosition > (5 * targetRect.height) / 6
+    }
+  }
+
+  function onDragOver (event: DragEvent, object: Ref<Document>): void {
+    event.preventDefault()
+    if (!(event.target as HTMLElement).draggable) return
+    if (event.dataTransfer === null || event.target === null || draggedItem === object) {
+      return
+    }
+
+    if (draggedItem !== undefined && canDrop(draggedItem, object)) {
+      event.dataTransfer.dropEffect = 'move'
+      draggedOver = object
+
+      const { before, after } = getDropPosition(event)
+      draggedOverBefore = before
+      draggedOverAfter = after
+    } else {
+      event.dataTransfer.dropEffect = 'none'
+    }
+  }
+
+  function onDragEnd (event: DragEvent): void {
```

Comment:
Please check a couple of cases where UX doesn't match what actually happens (shared descriptions offline)

Reply:
Improved UX for these cases